spec/linux/kernel.go

```diff
@@ -42,5 +43,72 @@ func (g *KernelGenerator) Generate() (interface{}, error) {
 		results[key] = str
 	}
 
+	hostInfo, err := host.Info()
+	if err != nil {
+		kernelLogger.Errorf("Failed to get platform information: %s", err)
```

Comment:
you should return when getting err. `hostInfo` can be nil.
 
`return nil, err`